cmd/snap/cmd_snap_op.go

```diff
@@ -144,15 +144,62 @@ func (x *cmdRemove) Execute([]string) error {
 	return nil
 }
 
+type channelMixin struct {
+	Channel string `long:"channel" description:"Use this channel instead of the device's default"`
+
+	// shortcuts
+	EdgeChannel      bool `long:"edge" description:"Install from the edge channel"`
+	BetaChannel      bool `long:"beta" description:"Install from the beta channel"`
+	CandidateChannel bool `long:"candidate" description:"Install from the candidate channel"`
+	StableChannel    bool `long:"stable" description:"Install from the stable channel"`
+}
+
+func (x *channelMixin) setChannel(channel string) error {
+	if x.Channel != "" {
+		return fmt.Errorf("Please specify a single channel")
+	}
+	x.Channel = channel
+	return nil
+}
+
+func (x *channelMixin) setChannelFromCommandline() error {
+	if x.StableChannel {
+		if err := x.setChannel("stable"); err != nil {
```

Comment:
Hm, not sure, would make it slightly shorter, but I think it would be `maybeSetChannel()` or something then because it will not always set the channel.